provider/manifest/manager.go

```diff
@@ -333,14 +341,58 @@ func (m *manager) validateRequests() {
 	}
 }
 
+var errManifestRejected = errors.New("manifest rejected")
+
+func (m *manager) checkHostnamesForManifest(requestManifest manifest.Manifest, groupNames []string) error {
+	allHostnames := make([]string, 0)
+
+	for _, mgroup := range requestManifest.GetGroups() {
+		for _, groupName := range groupNames {
+			// Only check leases with a matching deployment ID & group name
+			if groupName == mgroup.GetName() {
+				allHostnames = append(allHostnames, util.AllHostnamesOfManifestGroup(mgroup)...)
+				// For each service that exposes via an Ingress, then require a hsotname
+				if m.config.HTTPServicesRequireAtLeastOneHost {
+					for _, service := range mgroup.Services {
+						for _, expose := range service.Expose {
+							if util.ShouldBeIngress(expose) && len(expose.Hosts) == 0 {
+								return fmt.Errorf("%w: service %q exposed on %d:%s must have a hostname", errManifestRejected, service.Name, util.ExposeExternalPort(expose), expose.Proto)
+							}
+						}
+					}
+				}
+
+			}
+		}
+	}
+
+	// Check if the hostnames are available. Do not block forever
+	select {
+	case err := <-m.hostnameService.CanReserveHostnames(allHostnames, m.data.Deployment.DeploymentID):
+		return err
+	case <-m.lc.ShutdownRequest():
+		return ErrNotRunning
+	}
+
+}
+
 func (m *manager) validateRequest(req manifestRequest) error {
 	// ensure that an uploaded manifest matches the hash declared on
 	// the Akash Deployment.Version
 	version, err := sdl.ManifestVersion(req.value.Manifest)
 	if err != nil {
 		return err
 	}
-	if !bytes.Equal(version, m.data.Deployment.Version) {
+
+	var versionExpected []byte
```

Comment:
@boz @troian  I refactored the code here that checks the manifest version that is sent vs. what comes from the blockchain.

Previously, only `m.data.Deployment.Version` was checked. This was the first value received. The value `m.versions` was never checked. After this change, it checks the last version (which should be the most recent version broadcast on the blockchain) seen.

Reply:
Thanks.  This is screaming for a rewrite.